app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt

```diff
@@ -218,11 +219,19 @@ class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_da
         // and we want to prevent a crash by defaulting to pop home behavior
         if (checkIfFragmentInBackStack(R.id.deleteBrowsingDataFragment)) {
             // If Settings is in the backstack then we can continue with intended behavior
-            if (checkIfFragmentInBackStack(R.id.browserFragment)) findNavController().popBackStack(
-                R.id.homeFragment,
-                false
-            )
-            else findNavController().popBackStack()
+            if (checkIfFragmentInBackStack(R.id.browserFragment)) {
+                findNavController().popBackStack(
+                    R.id.homeFragment,
+                    false
+                )
+                while (findNavController().currentDestination?.id == R.id.homeFragment) {
```

Comment:
@ekager Is there someplace else we do something like this without using `while`?